<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Version 0.2 • acidplots</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Version 0.2" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-news">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">acidplots</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.26</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/acidplots.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/acidgenomics/acidplots/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small>0.2</small></h1>
      
    </div>

    <div id="acidplots-0-2-0-2019-04-22" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-0-2019-04-22" class="anchor"></a>acidplots 0.2.0 (2019-04-22)</h2>
<p>Reworked package, consolidating previous minimalism and firestarter packages.</p>
<div id="new-functions-2" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions-2" class="anchor"></a>New functions</h3>
<ul>
<li>
<code>synesthesia</code> color palette functions.</li>
<li>
<code>acid_coord_flip</code>: Intelligently puts samples at the top of the Y axis when flipped. Particularly useful for barplots.</li>
<li>
<code>acid_pretty_breaks</code>: Attempt to improve <code><a href="https://scales.r-lib.org//reference/breaks_pretty.html">scales::pretty_breaks</a></code>, using an approach more similar to the base plot engine.</li>
<li>Renamed ggplot2 themes: <code>acid_theme_light</code>, <code>acid_theme_dark</code>.</li>
</ul>
</div>
<div id="major-changes-6" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-6" class="anchor"></a>Major changes</h3>
<ul>
<li>All ggplot2 geoms and transformations (<code>ggproto</code> functions) now consistently use <code>acid_</code> as a prefix.</li>
</ul>
</div>
<div id="minor-changes-21" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-21" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>plotCountsPerBiotype</code> and <code>plotCountsPerBroadClass</code> now support <code>title</code>.</li>
</ul>
</div>
<div id="deprecations" class="section level3">
<h3 class="hasAnchor">
<a href="#deprecations" class="anchor"></a>Deprecations</h3>
<ul>
<li>
<code>plotCountsPerGene</code> has been renamed to <code>plotCountsPerFeature</code>.</li>
<li>
<code>plotGenesDetected</code> has been renamed to <code>plotFeaturesDetected</code>.</li>
<li>Deprecated <code>theme_paperwhite</code> and <code>theme_midnight</code> in favor of <code>acid_theme_light</code> and <code>acid_theme_dark</code>.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-1-2019-04-25" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-1-2019-04-25" class="anchor"></a>acidplots 0.2.1 (2019-04-25)</h2>
<div id="new-functions-1" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions-1" class="anchor"></a>New functions</h3>
<ul>
<li>New <code>gradient</code> palette functions: <code>purpleOrange</code>, <code>blueYellow</code>.</li>
<li>Remove gplots package from imports.</li>
</ul>
</div>
<div id="major-changes-5" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-5" class="anchor"></a>Major changes</h3>
<ul>
<li>
<code>plotHeatmap</code> now supports automatic color calculation when <code>breaks</code> argument is defined. Also improved support for <code>legendBreaks</code>.</li>
</ul>
</div>
<div id="minor-changes-20" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-20" class="anchor"></a>Minor changes</h3>
<ul>
<li>Fix <code>.pheatmapColorPalette</code> internal check for <code>color</code>, which requires <code>n</code> argument, rather than assuming positional variable.</li>
<li>
<code>plotCountsPerFeature</code>: Improve the count subtitle.</li>
<li>S4 generic reexport documentation fixes.</li>
<li>Miscellaneous documentation fixes.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-2-2019-04-26" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-2-2019-04-26" class="anchor"></a>acidplots 0.2.2 (2019-04-26)</h2>
<div id="major-changes-4" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-4" class="anchor"></a>Major changes</h3>
<ul>
<li>Now importing <code>upset</code> function from UpSetR, with modified formals, so we can easily call this from other packages, including <a href="https://deseqanalysis.acidgenomics.com/">DESeqAnalysis</a>.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-3-2019-05-05" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-3-2019-05-05" class="anchor"></a>acidplots 0.2.3 (2019-05-05)</h2>
<div id="major-changes-3" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-3" class="anchor"></a>Major changes</h3>
<ul>
<li>Now pinned against R &gt;= 3.5.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-4-2019-05-08" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-4-2019-05-08" class="anchor"></a>acidplots 0.2.4 (2019-05-08)</h2>
<div id="minor-changes-19" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-19" class="anchor"></a>Minor changes</h3>
<ul>
<li>Improved log2 and log10 axis scale handling for <code>plotCountsPerBiotype</code> and <code>plotCountsPerBroadClass</code>. Previously, the <code>breaks</code> argument was defined in internal <code>scale_y_continuous</code> ggplot2 call, but this doesn’t perform well for many RNA-seq datasets at log2 scale. Instead, it’s better to log2 scale the data first and then simplify indicate log2 on the y axis title.</li>
<li>
<code>plotCountsPerFeature</code>: Improved custom color handling for density geom. Removed violin geom support but added jitter geom support. The violin plot method doesn’t scale well for multiple samples, and the boxplot is visually easier to interpret for this metric, in general.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-5-2019-07-17" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-5-2019-07-17" class="anchor"></a>acidplots 0.2.5 (2019-07-17)</h2>
<div id="minor-changes-18" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-18" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>upset</code>: Compatibility update for UpSetR v1.4 release.</li>
<li>Compatibility fixes for basejump v0.10.11 release.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-6-2019-07-18" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-6-2019-07-18" class="anchor"></a>acidplots 0.2.6 (2019-07-18)</h2>
<div id="minor-changes-17" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-17" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>plotCountsPerBiotype</code>: Internal tidyeval bug fix for call to <code><a href="https://dplyr.tidyverse.org/reference/top_n.html">dplyr::top_n</a></code>. Previously, the <code>n</code> argument supported direct variable input, but now it must be unquoted, due to change in underlying rlang engine following the 0.4 update.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-7-2019-07-23" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-7-2019-07-23" class="anchor"></a>acidplots 0.2.7 (2019-07-23)</h2>
<div id="new-functions" class="section level3">
<h3 class="hasAnchor">
<a href="#new-functions" class="anchor"></a>New functions</h3>
<ul>
<li>
<code>plotCountsCorrelation</code> and <code>plotCountsCorrelationHeatmap</code>: New functions optimized for plotting correlation between gene expression matrices. Particularly useful for comparing pseudoaligned vs. aligned counts, which is implemented in the new bcbioRNASeq package update.</li>
</ul>
</div>
<div id="minor-changes-16" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-16" class="anchor"></a>Minor changes</h3>
<ul>
<li>Improved naming consistency of internal S4 functions.</li>
<li>Updated basejump dependencies.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-8-2019-07-29" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-8-2019-07-29" class="anchor"></a>acidplots 0.2.8 (2019-07-29)</h2>
<div id="minor-changes-15" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-15" class="anchor"></a>Minor changes</h3>
<ul>
<li>Improved warning suppression for partial match in <code>pheatmap</code> calls.</li>
<li>Improved documentation.</li>
<li>Improving automatic plot title and label handling with new <code>makeLabel</code> and <code>makeTitle</code> functions from syntactic update.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-9-2019-07-30" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-9-2019-07-30" class="anchor"></a>acidplots 0.2.9 (2019-07-30)</h2>
<div id="minor-changes-14" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-14" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated basejump dependency versions.</li>
<li>Updated some unit tests to reflect RangedSummarizedExperiment example update in acidtest package.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-10-2019-08-06" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-10-2019-08-06" class="anchor"></a>acidplots 0.2.10 (2019-08-06)</h2>
<div id="minor-changes-13" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-13" class="anchor"></a>Minor changes</h3>
<ul>
<li>Improved documentation consistency by importing shared parameter roxygen from the new acidroxygen package.</li>
<li>Updated basejump dependency versions.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-11-2019-08-12" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-11-2019-08-12" class="anchor"></a>acidplots 0.2.11 (2019-08-12)</h2>
<div id="major-changes-2" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-2" class="anchor"></a>Major changes</h3>
<ul>
<li>Migrated some QC plotting functions previously defined in bcbioSingleCell package. This makes it possible to share the code for new Chromium package, designed for importing single cell RNA-seq data from 10X Genomics.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-12-2019-08-20" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-12-2019-08-20" class="anchor"></a>acidplots 0.2.12 (2019-08-20)</h2>
<div id="minor-changes-12" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-12" class="anchor"></a>Minor changes</h3>
<ul>
<li>Improved consistency of internal tibble handling inside plot functions.</li>
<li>Updated basejump dependency versions.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-13-2019-08-21" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-13-2019-08-21" class="anchor"></a>acidplots 0.2.13 (2019-08-21)</h2>
<div id="minor-changes-11" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-11" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>plotPCA</code>: Improved automatic <code>ntop</code> argument handling when passing in a value greater than the number or features (rows) defined in the object.</li>
<li>Improved documentation for <code>SingleCellExperiment</code> methods that need to average or aggregate expression at sample level.</li>
<li>Removed dplyr and magrittr dependencies.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-14-2019-08-27" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-14-2019-08-27" class="anchor"></a>acidplots 0.2.14 (2019-08-27)</h2>
<div id="minor-changes-10" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-10" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>plotCountsPerFeature</code>: Renamed formals to match conventions in new basejump <code>melt</code> methods.</li>
<li>Updated basejump dependencies and NAMESPACE.</li>
<li>Tightened up if/then checks, ensuring boolean flag return.</li>
<li>Improved some working examples and unit tests.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-15-2019-09-13" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-15-2019-09-13" class="anchor"></a>acidplots 0.2.15 (2019-09-13)</h2>
<div id="major-changes-1" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes-1" class="anchor"></a>Major changes</h3>
<ul>
<li>
<code>acid_theme_dracula</code>: New dark mode theme supporting Dracula color palette.</li>
<li>Exporting color palette vectors: <code>lightPalette</code>, <code>darkPalette</code>, <code>draculaPalette</code>, <code>iOSLightPalette</code>, <code>iOSDarkPalette</code>, <code>macOSLightPalette</code>, <code>macOSDarkPalette</code>.</li>
<li>
<code>synesthesia</code>: Simplified color palette to use purple, blue, green, orange. User can tweak the palette using a new <code>palette</code> argument.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-16-2019-09-17" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-16-2019-09-17" class="anchor"></a>acidplots 0.2.16 (2019-09-17)</h2>
<div id="major-changes" class="section level3">
<h3 class="hasAnchor">
<a href="#major-changes" class="anchor"></a>Major changes</h3>
<ul>
<li>Replaced <code>title</code> argument with <code>labels</code> argument for functions that render plots using ggplot2.</li>
<li>Now exporting <code>matchLabels</code> function, that handles internal ggplot2 label matching, and auto-populates with defaults in formal argument.</li>
<li>Using “samplesAxis” and “countsAxis” labels for ggplot2 return that support multiple geoms. Consider a similar approach for QC metric plots.</li>
</ul>
</div>
<div id="minor-changes-9" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-9" class="anchor"></a>Minor changes</h3>
<ul>
<li>Simplified internal handoff from <code>SingleCellExperiment</code> S4 methods to <code>SummarizedExperiment</code>, using standard <code>...</code> passthrough instead of relying upon <code>matchArgsToDoCall</code>.</li>
<li>Improved color palette consistency in plotting functions.</li>
<li>
<code>plotCounts</code>: Moved <code>DESeqDataSet</code> method to DESeqAnalysis package.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-17-2019-11-07" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-17-2019-11-07" class="anchor"></a>acidplots 0.2.17 (2019-11-07)</h2>
<div id="minor-changes-8" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-8" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated dependencies to require Bioconductor 3.10 release.</li>
<li>Added return value for <code>matchLabels</code> in documentation.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-18-2019-11-12" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-18-2019-11-12" class="anchor"></a>acidplots 0.2.18 (2019-11-12)</h2>
<div id="minor-changes-7" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-7" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>plotCounts</code>: Reworked <code>medianLine</code> argument and renamed to simply <code>line</code>. Now supports mean and geometric mean in addition to median now. Line is now disabled by default.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-19-2019-11-19" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-19-2019-11-19" class="anchor"></a>acidplots 0.2.19 (2019-11-19)</h2>
<div id="minor-changes-6" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-6" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>plotCounts</code>: The <code>gene</code> argument is now optional, supporting subset <code>SummarizedExperiment</code> objects. Currently a maximum of 20 genes (rows) are allowed to pass through in the method.</li>
<li>Updated documentation to support roxygen2 7.0 release.</li>
<li>Improved default color palettes for <code>plotHeatmap</code>, <code>plotCorrelationHeatmap</code>, and <code>plotQuantileHeatmap</code>.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-20-2019-12-09" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-20-2019-12-09" class="anchor"></a>acidplots 0.2.20 (2019-12-09)</h2>
<div id="minor-changes-5" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-5" class="anchor"></a>Minor changes</h3>
<ul>
<li>Hardened user-defined <code>interestingGroups</code> input against <code>NA</code> values causing values to drop in plots. This was discovered with <code>plotPCA</code> calling <code>geom_point</code> internally, which caused removal of rows. If we coerce <code>NA</code> value to <code>"NA"</code> first, then removal no longer occurs. This is accomplished internally using <code>str_remove_na</code> (see also <code>stringi_remove_na</code>). The change applies to these functions: <code>plotCellCounts</code>, <code>plotCounts</code>, <code>plotCountsPerBiotype</code>, <code>plotCountsPerBroadClass</code>, <code>plotCountsPerFeature</code>, <code>plotFeaturesDetected</code>, <code>plotPCA</code>, <code>plotTotalCounts</code>, <code>plotZerosVsDepth</code>.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-21-2019-12-13" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-21-2019-12-13" class="anchor"></a>acidplots 0.2.21 (2019-12-13)</h2>
<div id="minor-changes-4" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-4" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>plotHeatmap</code>: Increased the default break range from <code>-2:2</code> up to <code>-3:3</code>. This helps improve the dynamic range a bit for some RNA-seq plots.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-22-2020-01-20" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-22-2020-01-20" class="anchor"></a>acidplots 0.2.22 (2020-01-20)</h2>
<div id="minor-changes-3" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-3" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated bioverbs dependency to renamed acidgenerics package.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-23-2020-02-22" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-23-2020-02-22" class="anchor"></a>acidplots 0.2.23 (2020-02-22)</h2>
<div id="minor-changes-2" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-2" class="anchor"></a>Minor changes</h3>
<ul>
<li>Changed license from MIT to GPL-3.</li>
<li>Updated internal code to use <code>aggregateCellsToSamples</code> instead of <code>pseudobulk</code>, which has migrated to pointillism and will be reworked in an upcoming release.</li>
<li>Updated documentation for <code>SingleCellExperiment</code> methods, where <code>aggregateCellsToSamples</code> is called internally. This was propagating to other packages where <code>SingleCellExperiment</code> methods are not defined, making the documentation confusing.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-24-2020-03-11" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-24-2020-03-11" class="anchor"></a>acidplots 0.2.24 (2020-03-11)</h2>
<div id="minor-changes-1" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes-1" class="anchor"></a>Minor changes</h3>
<ul>
<li>
<code>acid_coord_flip</code>: Fix for breaking change introduced by ggplot2 v3.3.0, which renamed <code>object[["layers"]][[1L]][["geom"]][["required_aes"]]</code> from “x” to “x|y”. The package now requires this new version of ggplot2.</li>
</ul>
</div>
</div>
    <div id="acidplots-0-2-25-2020-04-13" class="section level2">
<h2 class="hasAnchor">
<a href="#acidplots-0-2-25-2020-04-13" class="anchor"></a>acidplots 0.2.25 (2020-04-13)</h2>
<div id="minor-changes" class="section level3">
<h3 class="hasAnchor">
<a href="#minor-changes" class="anchor"></a>Minor changes</h3>
<ul>
<li>Updated documentation and roxygen dependency to v7.1.0.</li>
<li>Updated ggplot2 and other graphical R package dependencies.</li>
<li>Fixed <code>plotHeatmap</code> unit tests to harden against <code>SummarizedExperiment</code> and <code>SingleCellExperiment</code> objects defined in acidtest package that contain rows and/or columns with all zeros. Refer to <code>nonzeroRowsAndCols</code> function in basejump for details.</li>
<li>Rebuilt pkgdown website using v1.5 update.</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>

</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://acidgenomics.com/'>Acid Genomics</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


