<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Parameters — params • acidplots</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  
<meta property="og:title" content="Parameters — params" />

<meta property="og:description" content="Parameters" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">acidplots</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/acidplots.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/acidgenomics/acidplots">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Parameters</h1>
    <small class="dont-index">Source: <a href='https://github.com/acidgenomics/acidplots/blob/master/R/params.R'><code>R/params.R</code></a></small>
    <div class="hidden name"><code>params.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Parameters</p>
    
    </div>

    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>aspectRatio</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Aspect ratio.</p></td>
    </tr>
    <tr>
      <th>assay</th>
      <td><p><code><a href='https://rdrr.io/r/base/vector.html'>vector(1)</a></code>.
Name or index of count matrix slotted in
<code><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assays()</a></code>. When passing in a string, the
name must be defined in
<code><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>assayNames()</a></code>.</p></td>
    </tr>
    <tr>
      <th>borderColor</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code> or <code>NULL</code>.
Border color. Disabled by default for improved aesthetics.</p></td>
    </tr>
    <tr>
      <th>color</th>
      <td><p><code>ScaleDiscrete</code>.
Desired ggplot2 color scale. Must supply discrete values. When set <code>NULL</code>,
the default ggplot2 color palette will be used. If manual color definitions
are desired, we recommend using <code><a href='https://rdrr.io/pkg/ggplot2/man/scale_manual.html'>ggplot2::scale_color_manual()</a></code>.</p>
<p>To set the discrete color palette globally, use:</p><pre>options(acid.color.discrete = ggplot2::scale_color_viridis_d())
</pre></td>
    </tr>
    <tr>
      <th>countsAxisLabel</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Counts axis label.</p></td>
    </tr>
    <tr>
      <th>dark</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Plot against a dark background using the
<code><a href='deprecated.html'>theme_midnight()</a></code> ggplot2 theme.</p></td>
    </tr>
    <tr>
      <th>dimsUse</th>
      <td><p><code>integer</code>.
Vector of length 2 that denotes the columns from the reduced dimension
matrix to use for <code>centerX</code> and <code>centerY</code> column calculations. Defaults the
first and second dimensions.</p></td>
    </tr>
    <tr>
      <th>direction</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Plot "<code>both</code>", "<code>up</code>", or "<code>down</code>" directions.</p></td>
    </tr>
    <tr>
      <th>expression</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Calculation to apply.
Uses <code><a href='https://rdrr.io/r/base/match.arg.html'>match.arg()</a></code> internally and defaults to the first
argument in the <code>character</code> vector.</p></td>
    </tr>
    <tr>
      <th>fill</th>
      <td><p><code>ggproto</code>/<code>ScaleDiscrete</code>.
Desired ggplot2 fill scale. Must supply discrete values. When set to
<code>NULL</code>, the default ggplot2 color palette will be used. If manual color
definitions are desired, we recommend using <code><a href='https://rdrr.io/pkg/ggplot2/man/scale_manual.html'>ggplot2::scale_fill_manual()</a></code>.</p>
<p>To set the discrete fill palette globally, use:</p><pre>options(acid.fill.discrete = ggplot2::scale_fill_viridis_d())
</pre></td>
    </tr>
    <tr>
      <th>flip</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Flip x and y axes. Recommended for plots containing many samples.</p></td>
    </tr>
    <tr>
      <th>gene2symbol</th>
      <td><p><code>Gene2Symbol</code>.
Gene-to-symbol mappings. Must contain <code>geneID</code> and <code>geneName</code> columns. See
<code>Gene2Symbol</code> for more information.</p></td>
    </tr>
    <tr>
      <th>genes</th>
      <td><p><code>character</code>.
Gene identifiers. It is considered better practice to input the stable gene
identifiers from Ensembl (e.g. "ENSG00000000003") and not the (HGNC) gene
symbols (e.g. "TSPN6"), if possible.</p></td>
    </tr>
    <tr>
      <th>geom</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Plot type. Uses <code><a href='https://rdrr.io/r/base/match.arg.html'>match.arg()</a></code> internally and defaults to
the first argument in the <code>character</code> vector.</p></td>
    </tr>
    <tr>
      <th>headerLevel</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code> (<code>1</code>-<code>7</code>).
Markdown header level.</p></td>
    </tr>
    <tr>
      <th>interestingGroups</th>
      <td><p><code>character</code>.
Groups of interest that define the samples. If left unset, defaults to
<code>sampleName</code>.</p></td>
    </tr>
    <tr>
      <th>label</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Superimpose sample text labels on the plot.</p></td>
    </tr>
    <tr>
      <th>labelSize</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Size of the text label.</p></td>
    </tr>
    <tr>
      <th>legend</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Show plot legend.</p></td>
    </tr>
    <tr>
      <th>level</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Return as genes or transcripts.</p></td>
    </tr>
    <tr>
      <th>limit</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code>.
Threshold to denote on the plot, using a dashed line.</p></td>
    </tr>
    <tr>
      <th>max</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code>.
Recommended maximum value cutoff.</p></td>
    </tr>
    <tr>
      <th>min</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code>.
Recommended minimum value cutoff.</p></td>
    </tr>
    <tr>
      <th>minCounts</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Minimum number of counts per gene in the count matrix.</p></td>
    </tr>
    <tr>
      <th>n</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Number to include.</p></td>
    </tr>
    <tr>
      <th>ntop</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Number of top genes to label.</p></td>
    </tr>
    <tr>
      <th>object</th>
      <td><p>Object.</p></td>
    </tr>
    <tr>
      <th>perMillion</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Display as counts per million.</p></td>
    </tr>
    <tr>
      <th>perSample</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Visualize the distributions per sample.</p></td>
    </tr>
    <tr>
      <th>plotlist</th>
      <td><p><code>list</code>.
List containing <code>ggplot</code> objects.</p></td>
    </tr>
    <tr>
      <th>pointAlpha</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code> (<code>0</code>-<code>1</code>).
Alpha transparency level. Useful when there many points in the dataset
(e.g. single-cell data), and some points can be masked.</p></td>
    </tr>
    <tr>
      <th>pointColor</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Default point color for the plot.</p></td>
    </tr>
    <tr>
      <th>pointScalar</th>
      <td><p><code><a href='https://rdrr.io/r/base/integer.html'>integer(1)</a></code>.
Default point size for the plot.</p></td>
    </tr>
    <tr>
      <th>pointsAsNumbers</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Plot the points as numbers (<code>TRUE</code>) or dots (<code>FALSE</code>).</p></td>
    </tr>
    <tr>
      <th>pointSize</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code>.
Point size for dots in the plot.</p></td>
    </tr>
    <tr>
      <th>return</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Return type. Uses <code><a href='https://rdrr.io/r/base/match.arg.html'>match.arg()</a></code> internally and defaults
to the first argument in the <code>character</code> vector.</p></td>
    </tr>
    <tr>
      <th>scale</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Whether the values should be centered and scaled in either the row or
column direction ("<code>row</code>", "<code>column</code>"), or remain unscaled ("<code>none</code>").</p></td>
    </tr>
    <tr>
      <th>subtitle</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Plot subtitle.</p></td>
    </tr>
    <tr>
      <th>title</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Plot title.</p></td>
    </tr>
    <tr>
      <th>trans</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Name of the axis scale transformation to apply.</p>
<p>For more information:</p><pre>help(topic = "scale_x_continuous", package = "ggplot2")
</pre></td>
    </tr>
    <tr>
      <th>trendline</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Include trendline on plot.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>No value.</p>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
          </ul>

  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://acidgenomics.com/'>Acid Genomics</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.9100.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


